app/customer.py

```diff
@@ -0,0 +1,76 @@
+import math
+from dataclasses import dataclass
+
+
+from app.shop import Shop, print_check
+
+
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list
+    money: int
+    car: dict
+
+
+def cost_of_km(customer: Customer, data: dict) -> int | float:
+    return (customer.car["fuel_consumption"] / 100) * data["FUEL_PRICE"]
+
+
+def cost_by_distance(
+        customer: Customer,
+        shop: Shop,
+        data: dict
+) -> int | float:
+    return (math.dist(
+        customer.location, shop.location
+    ) * 2) * cost_of_km(customer, data)
+
+
+def cost_of_products(customer: Customer, shop: Shop) -> int | float:
+    return sum(
+        [shop.products[product] * number
+         for product, number in customer.product_cart.items()]
+    )
+
+
+def total_price(customer: Customer, shop: Shop, data: dict) -> int | float:
+    return cost_by_distance(
+        customer, shop, data) + cost_of_products(
+        customer, shop)
+
+
+def trip_and_purchase(data: dict) -> None:
+    for customer_data in data["customers"]:

```

Comment:
The part of logic that creates instances of Customer and Shop can be moved to main.py. There you can create instances and then call methods to perform a shop trip on a customer instance